.github/workflows/teams.yml

```diff
@@ -0,0 +1,45 @@
+# A workflow to run the generate_teams.py script in the CI weekly and create a pull request
+
+on:
+  schedule:
+    # Run every Monday at 03:00 AM UTC
+    - cron: "0 3 * * 1"
+  workflow_dispatch:
+
+jobs:
+  generate-teams:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout main branch of repository
+        uses: actions/checkout@v3
+        with:
+          ref: main
+          submodules: recursive
+
+      - name: Setup Python 3.11 and run the teams generation script
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+      - run: python scripts/generate_teams.py
```

Comment:
The `uses` and `run` lines here will I think be 2 different steps. We might as well not club the name of the `run` step with a different step above (`uses`).

Reply:
I just copied the syntax from the [basic usage](https://github.com/actions/setup-python#basic-usage) section of the documentation, maybe it should work? Or should we add it in a different step because it will be cleaner?